src/models/activity.ts

```diff
@@ -0,0 +1,42 @@
+import {DynamoDbImage, parseStringArray} from "../services/dynamodb-images";
+import {debugLog} from "../services/logger";
+
+export interface Activity {
+    id?: string;
+    activityType?: string;
+    activityDay?: string;
+    startTime?: string;
+    endTime?: string;
+    testerStaffId?: string;
+    testerName?: string;
+    testerEmail?: string;
+    testStationType?: string;
+    testStationName?: string;
+    testStationPNumber?: string;
+    testStationEmail?: string;
+    parentId?: string;
+    notes?: string;
+    waitReason?: string[];
+}
+
+export const parseActivity = (image: DynamoDbImage): Activity => {
+    debugLog("Parsing activity...");
+
+    return {
+        id: image.getString("id"),
+        activityType: image.getString("activityType"),
+        activityDay: image.getString("activityDay"),
```

Comment:
this attribute is represented by DATE type in the database

Reply:
That's just YYYY-MM-DD isn't it?

Reply:
I think that's what it comes in as, so it can stay as string, cause there's no conversion to do

Reply:
From API Schema
`activityDay:
  type: string
  format: date
  nullable: true
`
All dates are of the string type in test result / tech records / activity, if you leave it as they are there will be implicit type casting as in database this field is DATE.

Reply:
Dates are strings formatted as ISO8601 in our micro services with cases where have legacy data (that we are aware of) which could contain up to 6 digits for the ms. (calculation of expiry date for certificates)
```js
 public static isValidDate(
    input: string | Date | number | undefined
  ): boolean {
    return (
      input !== undefined &&
      (moment(input, "YYYY-MM-DDTHH:mm:ss.SSSSSSZ", true).isValid() || // Legacy format
        moment(input, "YYYY-MM-DDTHH:mm:ss.SSSZ", true).isValid() || //  Test result API format
        moment(input, "YYYY-MM-DD", true).isValid()) && // Tech Record dates e.g. regnDate
      moment(input).isAfter(new Date(0))
    );
  }
  ```

Reply:
This is good to know, as it was a black-box before for us. API schema is sketchy sometimes and does not explicitly describe what format is expected. I would insist on explicit casting instead of treating it as a string.

Reply:
Remove `activity_day` column from DB and remove references from `update-store` as well.